pastis/matrix_generation/matrix_from_efields.py

```diff
@@ -134,6 +141,44 @@ def setup_single_mode_function(self):
         self.calculate_one_mode = functools.partial(_luvoir_matrix_single_mode, self.number_all_modes, self.wfe_aber,
                                                     self.luvoir, self.resDir, self.save_efields, self.saveopds)
 
+class MatrixEfieldRST(PastisMatrixEfields):
+    """
+    Class to calculate RST Electrical field of RST CGI.
+    """
+    instrument = 'RST'
+
+    def __init__(self, max_local_zernike=3, initial_path='', saveefields=True, saveopds=True):
+        super().__init__(initial_path=initial_path, saveefields=saveefields, saveopds=saveopds)
+        self.max_local_zernike = max_local_zernike
+
+    def calculate_ref_efield(self):
+        iwa = CONFIG_PASTIS.getfloat('RST', 'IWA')
+        owa = CONFIG_PASTIS.getfloat('RST', 'OWA')
+        self.rst_cgi = webbpsf_imaging.set_up_cgi()
+
+        # Calculate direct reference images for contrast normalization
+        rst_direct = self.rst_cgi.raw_PSF()
+        direct = rst_direct.calc_psf(nlambda=1, fov_arcsec=1.6)
+        direct_psf = direct[0].data
+        self.norm = direct_psf.max()
+
+        # Calculate dark hole mask
+        self.rst_cgi.working_area(im=direct_psf, inner_rad=iwa, outer_rad=owa)
+        self.dh_mask = self.rst_cgi.WA
+
+        # Calculate reference E-field in focal plane, without any aberrations applied
+        _trash, inter = self.rst_cgi.calc_psf(nlambda=1, fov_arcsec=1.6, return_intermediates=True)
+        self.efield_ref = inter[6].wavefront
+        # [6] is the last optic = detector
```

Comment:
Please move this comment to the line it belongs to, since it seems to be an in-line comment.